File: ts/packages/knowPro/src/answerResponseSchema.ts
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

export type AnswerType =
    | "NoAnswer" // If question cannot be accurately answered from [ANSWER CONTEXT]
    | "Answered"; // Fully answer question

export type AnswerResponse = {
    // use "NoAnswer" if no highly relevant answer found in the [ANSWER CONTEXT]
    type: AnswerType;
    // The question being answered
    question?: string;
    // the answer to display if [ANSWER CONTEXT] is highly relevant and can be used to answer the user's question
    answer?: string | undefined;
    // If NoAnswer, explain why..
    // particularly explain why you didn't use any supplied entities
    whyNoAnswer?: string;
};

File: python/ta/typeagent/demo/answer_response_schema.py
# Copyright (c) Microsoft Corporation.
# Licensed under the MIT License.

from typing import Literal, Annotated
from typing_extensions import Doc
from pydantic.dataclasses import dataclass

AnswerType = Literal[
    "NoAnswer",  # If question cannot be accurately answered from [ANSWER CONTEXT]
    "Answered",  # Fully answer question
]


@dataclass
class AnswerResponse:
    type: Annotated[
        AnswerType,
        Doc(
            'use "NoAnswer" if no highly relevant answer found in the [ANSWER CONTEXT]'
        ),
    ]
    answer: Annotated[
        str | None,
        Doc(
            "the answer to display if [ANSWER CONTEXT] is highly relevant and can be used to answer the user's question"
        ),
    ] = None
    whyNoAnswer: Annotated[
        str | None,
        Doc(
            "If NoAnswer, explain why..\nparticularly explain why you didn't use any supplied entities"
        ),
    ] = None
